lib/mcapi/crypto/jwe-crypto.js

```diff
@@ -135,6 +136,9 @@ function JweCrypto(config) {
         decryptionEncoding = "AES-128-CBC";
         secretKey = secretKey.slice(16, 32);
         gcmMode = false;
+        if(this.enableHmacVerification) {
+          verifyCbcHmac(jweTokenParts[0], iv, encryptedText, authTag, secretKey);
```

Comment:
The secretKey is incorrect here I think, as it has already been sliced to be the encryption key. So the mac should be extracted and passed here. Something like:

```suggestion
const macKey = secretKey.slice(0, 16);
secretKey = secretKey.slice(16, 32);
gcmMode = false;
        if(this.enableHmacVerification) {
          verifyCbcHmac(jweTokenParts[0], iv, encryptedText, authTag, macKey);
```